src/ripple/nodestore/impl/Shard.h

```diff
@@ -88,44 +90,50 @@ class Shard
     void
     sweep();
 
-    bool
-    validate();
-
     std::uint32_t
-    index() const {return index_;}
+    index() const
+    {
+        return index_;
+    }
+
+    std::shared_ptr<Backend> const&
+    getBackend() const;
 
     bool
-    complete() const {assert(backend_); return complete_;}
+    complete() const;
 
-    std::shared_ptr<PCache>&
-    pCache() {assert(backend_); return pCache_;}
+    std::shared_ptr<PCache>
+    pCache() const;
 
-    std::shared_ptr<NCache>&
-    nCache() {assert(backend_); return nCache_;}
+    std::shared_ptr<NCache>
+    nCache() const;
 
     std::uint64_t
-    fileSize() const {assert(backend_); return fileSz_;}
+    fileSize() const;
 
     std::uint32_t
-    fdRequired() const {assert(backend_); return fdRequired_;}
-
-    std::shared_ptr<Backend> const&
-    getBackend() const {assert(backend_); return backend_;}
+    fdRequired() const;
 
     std::shared_ptr<Ledger const>
-    lastStored() {assert(backend_); return lastStored_;}
+    lastStored() const;
+
+    bool
+    validate();
 
 private:
     friend class boost::serialization::access;
-    template<class Archive>
-    void serialize(Archive & ar, const unsigned int version)
+
+    template <class Archive>
+    void
+    serialize(Archive & ar, const unsigned int version)
```

Comment:
As `serialize(ar, version)` is currently implemented I think it's not thread safe.  I suspect it needs a `lock`.  But that's okay, it also looks like it is unused.  So my suggestion would be to delete the private method.

Reply:
Its a remnant and can be removed. Good find.